graphbolt/include/graphbolt/csc_sampling_graph.h

```diff
@@ -232,6 +238,34 @@ torch::Tensor Pick(
     int64_t offset, int64_t num_neighbors, int64_t fanout, bool replace,
     const torch::TensorOptions& options);
 
+/**
+ * @brief Picks a specified number of neighbors for a node per edge type,
+ * starting from the given offset and having the specified number of neighbors.
+ *
+ * @param offset The starting edge ID for the connected neighbors of the sampled
+ * node.
+ * @param num_neighbors The number of neighbors to pick.
+ * @param fanouts The number of edges to be sampled for each node per edge type.
+ * The value of it should be greater than or equal to 0 or -1. If -1 is given,
+ * it is equivalent to when the fanout is greater or equal to the number of
+ * eighbors and replacement is false, in which case all the neighbors will be
+ * selected. In contrast, a non-negative integer determines the minimum number
+ * of neighbors to select. The minimum value is determined by comparing the
+ * fanout value with the total number of neighbors available.
+ * @param replace Boolean indicating whether the sample is preformed with or
+ * without replacement. If True, a value can be selected multiple
+ * times.Otherwise, each value can be selected only once.
+ * @param options Tensor options specifying the desired data type of the result.
+ * @param type_per_edge Tensor representing the type of each edge in the
+ * original graph.
+ *
+ * @return A tensor containing the picked neighbors.
+ */
+torch::Tensor PickEtype(
```

Comment:
```suggestion
torch::Tensor EtypeBasedPick(
```

Reply:
Or PickByEType

Reply:
Fixed, use `PickByEtype` as it sounds more accurate.